adlfs/lib.py

```diff
@@ -198,6 +228,7 @@ def call(self, op, path='', **kwargs):
         """
         if op not in self.ends:
             raise ValueError("No such op: %s", op)
+        self._check_token()
```

Comment:
We check the expiry time on every request, not explicitly that the token is valid in some other sense. The expiry test has 100s grace built-in (out of the typical validity time of 3600s), so the refresh should always happen before expiry.
If multiple threads are launched with a nearly-expired token, it is conceivable that they will each refresh simultaneously - I don't think this is likely to ever happen, but maybe it would leads to undefined behaviour.